jax/_src/lax/windowed_reductions.py

```diff
@@ -19,50 +19,60 @@
 from typing import Callable
 import warnings
 
-import numpy as np
-
 from jax import tree_util
-
 from jax._src import ad_util
 from jax._src import core
 from jax._src import dispatch
 from jax._src import dtypes
 from jax._src import util
-from jax._src.core import ShapedArray, ConcreteArray
+from jax._src.core import ConcreteArray, ShapedArray
 from jax._src.interpreters import ad
 from jax._src.interpreters import batching
 from jax._src.interpreters import mlir
-from jax._src.lax import lax
 from jax._src.lax import convolution
+from jax._src.lax import lax
 from jax._src.lax import slicing
 from jax._src.lib.mlir import ir
 from jax._src.lib.mlir.dialects import hlo
 from jax._src.numpy.ufuncs import logaddexp
 from jax._src.typing import Array
+import numpy as np
 
 map = util.safe_map
 zip = util.safe_zip
 
 
-def reduce_window(operand, init_value, computation: Callable,
-                  window_dimensions: core.Shape, window_strides: Sequence[int],
-                  padding: str | Sequence[tuple[int, int]],
-                  base_dilation: Sequence[int] | None = None,
-                  window_dilation: Sequence[int] | None = None) -> Array:
+def _reduce_window(
+    operand,
+    init_value,
+    computation,
+    window_dimensions: core.Shape,
+    window_strides: Sequence[int],
+    padding: str | Sequence[tuple[int, int]],
+    base_dilation: Sequence[int] | None = None,
+    window_dilation: Sequence[int] | None = None,
+    is_jvp=False,
+):
   """Wraps XLA's `ReduceWindowWithGeneralPadding
   <https://www.tensorflow.org/xla/operation_semantics#reducewindow>`_
   operator.
   """
   flat_operands, operand_tree = tree_util.tree_flatten(operand)
   flat_init_values, init_value_tree = tree_util.tree_flatten(init_value)
-  if operand_tree != init_value_tree:
-    raise ValueError('Operands must have the same tree structure as '
-                     f'init_values: {operand_tree} vs. {init_value_tree}')
+  if not is_jvp:
+    if operand_tree != init_value_tree:
+      raise ValueError(
+          "Operands must have the same tree structure as "
+          f"init_values: {operand_tree} vs. {init_value_tree}"
```

Comment:
Separate from this PR, we should use our nicer tree mismatch error reporting utilities here, like tree_util.equality_errors and/or tree_util.prefix_errors.